### PR TITLE
WIP - refactor zscore code

### PR DESCRIPTION
**Pull request by gergness**
Use the `*_bases` functions to unify the calculation of zscores. I haven't had time to write unit tests yet, but I think this simplifies the zscore code and will make it easier for me to further refactor after the `rows_margin` consolidation.

It does have 2 errors in exporter though, but the only difference is whether there's a single hyphen or multiple. For example:

```
E       -              7   First Mov a2           58.3           ----          100.0              A              -              -
E       +              7   First Mov a2           58.3           ----          100.0              A              -           ----
```

Definitely need to write unit tests, but wanted to send my progress so far so that you can look during your day. Maybe we can cafe when I get on tomorrow.

### CODE REVIEW

**Comment by ernestoarbitrio** [comment]
> Use the `*_bases` functions to unify the calculation of zscores. I haven't had time to write unit tests yet, but I think this simplifies the zscore code and will make it easier for me to further refactor after the `rows_margin` consolidation.
> 
> It does have 2 errors in exporter though, but the only difference is whether there's a single hyphen or multiple. For example:
> 
> ```
> E       -              7   First Mov a2           58.3           ----          100.0              A              -              -
> E       +              7   First Mov a2           58.3           ----          100.0              A              -           ----
> ```
> 
> Definitely need to write unit tests, but wanted to send my progress so far so that you can look during your day. Maybe we can cafe when I get on tomorrow.

I think the errors are not releted to your changes but if you start your branch from master that includes my hypothesis testing changes that's the reason of the hyphens thing

**Review by ernestoarbitrio — CHANGES_REQUESTED** [review]
Looks good overall, just couple of comments

File: src/cr/cube/matrix/measure.py [hunk 1]
```diff
@@ -1564,16 +1563,92 @@ def blocks(self):
...
+        This is the third "block" and has the shape (n_row_subtotals, n_cols).
+        """
+        return self._zscore_calculation(
+            self._second_order_measures.weighted_counts.blocks[1][0],
+            self._second_order_measures.table_weighted_bases.blocks[1][0],
+            self._second_order_measures.row_weighted_bases.blocks[1][0],
+            self._second_order_measures.column_weighted_bases.blocks[1][0],
+        )
+
+    def _zscore_calculation(self, counts, total, rowsum, colsum):
```
[the remark this answers is not in the document]
Comment (gergness) [reply src/cr/cube/matrix/measure.py:1640]: What do you think of `_calculate_zscores`? I feel like `_zscores` sounds like a property not a method?

[the remark this answers is not in the document]
Comment (ernestoarbitrio) [reply src/cr/cube/matrix/measure.py:1640]: yeah like it

**Review by ernestoarbitrio — APPROVED** [review]
LGTM

**Pull request merged**